lib/v2/chinawriter/index.js

```diff
@@ -0,0 +1,83 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const iconv = require('iconv-lite');
+const jschardet = require('jschardet');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+
+const decodeBuffer = (buffer) => iconv.decode(buffer, jschardet.detect(buffer).encoding);
+
+module.exports = async (ctx) => {
+    const { id } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 5;
+
+    const rootUrl = 'http://www.chinawriter.com.cn';
+    const currentUrl = new URL(id ?? '', rootUrl).href;
+
+    const { data: response } = await got(currentUrl, {
+        responseType: 'buffer',
+    });
+
+    const $ = cheerio.load(decodeBuffer(response));
+
+    let items = $('main div.inner')
+        .find('a')
+        .toArray()
+        .filter((item) => /\/\d{4}\/\d{4}\/\w+-\w+\.html/.test($(item).prop('href')))
+        .slice(0, limit)
+        .map((item) => {
+            item = $(item);
+
+            const link = item.prop('href');
+
+            return {
+                title: item.text(),
+                link: /^http/.test(link) ? link : new URL(item.prop('href'), rootUrl).href,
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: detailResponse } = await got(item.link, {
+                    responseType: 'buffer',
+                });
+
+                const content = cheerio.load(decodeBuffer(detailResponse));
+
+                content('div.end_shared').remove();
+
+                const info = content('div.end_info').text().trim();
+
+                item.title = content('#newstit, h6.end_tit').text();
+                item.description = content('div.end_article').html();
+                item.author = info.match(/\|(.*)\d{4}/)[1].trim();
```

Comment:
This regex may not always works.
e.g., `http://www.chinawriter.com.cn/n1/2023/0303/c404928-32635716.html` from `/chinawriter/403978`